pkg/csi/service/node.go

```diff
@@ -381,21 +381,22 @@ func (driver *vsphereCSIDriver) NodeGetInfo(
 		}
 	}
 
-	if cnstypes.CnsClusterFlavor(os.Getenv(csitypes.EnvClusterFlavor)) == cnstypes.CnsClusterFlavorGuest {
-		nodeInfoResponse = &csi.NodeGetInfoResponse{
-			NodeId:             nodeID,
-			MaxVolumesPerNode:  maxVolumesPerNode,
-			AccessibleTopology: &csi.Topology{},
-		}
-		log.Infof("NodeGetInfo response: %v", nodeInfoResponse)
-		return nodeInfoResponse, nil
-	}
-
 	var (
 		accessibleTopology map[string]string
 	)
-	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.ImprovedVolumeTopology) {
-		// Initialize volume topology service.
+
+	if cnstypes.CnsClusterFlavor(os.Getenv(csitypes.EnvClusterFlavor)) == cnstypes.CnsClusterFlavorGuest {
+		if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TKGsHA) {
+			nodeInfoResponse = &csi.NodeGetInfoResponse{
+				NodeId:             nodeID,
+				MaxVolumesPerNode:  maxVolumesPerNode,
+				AccessibleTopology: &csi.Topology{},
+			}
+			log.Infof("NodeGetInfo response: %v", nodeInfoResponse)
+			return nodeInfoResponse, nil
+		}
+
+		// Initialize volume topology service if tkgs-ha is enabled in guest cluster.
```

Comment:
where are we fetching node topology labels here in this block?

I do not see calling `accessibleTopology, err = topologyService.GetNodeTopologyLabels(ctx, &nodeInfo)` for clustertype == guest && TKGsHA enabled.

Reply:
I see it at line 408, ignore this